test/test_other.py

```diff
@@ -1307,6 +1307,45 @@ def test_export_all(self):
     self.emcc('lib.c', ['-Oz', '-sEXPORT_ALL', '-sLINKABLE', '--pre-js', 'main.js'], output_filename='a.out.js')
     self.assertContained('libf1\nlibf2\n', self.run_js('a.out.js'))
 
+  def test_export_keepalive(self):
+    create_file('main.c', r'''
+      #include <emscripten.h>
+      EMSCRIPTEN_KEEPALIVE int libf1() { return 42; }
+    ''')
+
+    create_file('pre.js', '''
+      Module.onRuntimeInitialized = () => {
+        console.log(Module._libf1 ? Module._libf1() : 'unexported');
+      };
+    ''')
+
+    # By default, all kept alive functions should be exported.
+    self.do_runf('main.c', '42\n', emcc_args=['--pre-js', 'pre.js'])
+
+    # Ensures that EXPORT_KEEPALIVE=0 remove the exports
+    self.do_runf('main.c', 'unexported\n', emcc_args=['-sEXPORT_KEEPALIVE=0', '--pre-js', 'pre.js'])
+
+  def test_minimal_modularize_export_keepalive(self):
+    create_file('main.c', r'''
+      #include <emscripten.h>
+      EMSCRIPTEN_KEEPALIVE int libf1() { return 42; }
+    ''')
+
+    # With MINIMAL_RUNTIME, the module isn't exported.
```

Comment:
I'm not quite sure that this is saying here.    Are we trying to say, "The Module object doesn't exist under MINIMAL_RUNTIME, so exporting only makes sense alongside `-sMODULARIZE`"?

Reply:
The function basically isn't exported as a UMD nor ES6, so you can't require it. This is another issue with MINIMAL_RUNTIME :D

```js
var MyModuleName = (function(){ return ... })();
// Missing module.exports here 
```

We should probably also add a `-sEXPORT_UMD` like `-sEXPORT_ES6` but that's not for this PR. I didn't use `EXPORT_ES6` because then I need the same fix as the one in `test_minimal_runtime_export_all_modularize` to run in node, it's not worth it for such a simple thing.

In the meantime, I improved the comment.